src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.cs

```diff
@@ -677,5 +677,26 @@ public async Task When_ReadOnly_Toggled_Repeatedly()
 
 			Assert.AreEqual(updatedText.Length, textBox.SelectionLength);
 		}
+
+#if HAS_UNO
+		[TestMethod]
+		public async Task When_Recycling()
+		{
+			using var _ = FeatureConfigurationHelper.UseTemplatePooling();
+
+			var text = "Hello";
+			var textBox = new TextBox() { Text = text };
+			textBox.OnTemplateRecycled();
+
+			Assert.IsTrue(string.IsNullOrEmpty(textBox.Text));
+
+			textBox.IsTemplateRecyclingEnabled = false;
+
+			textBox.Text = "Hello";
+			textBox.OnTemplateRecycled();
+
+			Assert.AreEqual(text, textBox.Text);
```

Comment:
This is not the original issue right?

I mean it's totally okay for textBox.Text to be empty here, but **the way** the text changed has unintended side effects, is that right? If so, do we have a way to reproduce the unintended side effects in the test?

At the very least, the test needs an explanation comment.

What I would like to do is have a new way of changing property value that doesn't push two way bindings.

That would probably be a boolean property on `IFrameworkTemplatePoolAware` that gets set and unset before and after [this line](https://github.com/unoplatform/uno/blob/79887869b418b12d7fcb17730a27af753b9adb7c/src/Uno.UI/UI/Xaml/FrameworkTemplatePool.cs#L349C26-L349C42).

Then somewhere in the binding engine have check along the lines of:

```csharp
if (ActualInstance is IFrameworkTemplatePoolAware templateAwareElement && templateAwareElement.IsRecycling)
{
    // skip pushing bindings when properties change.
}
```